payments-core/src/main/java/com/stripe/android/paymentsheet/viewmodels/PaymentDataCollectionViewModel.kt

```diff
@@ -0,0 +1,21 @@
+package com.stripe.android.paymentsheet.viewmodels
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+
+internal class PaymentDataCollectionViewModel(
```

Comment:
This is the ViewModel that the fragments will share, so they don't need to know about the `BaseSheetViewModel`. I played a little with what would belong here. The fragments need to be able to set the `PaymentSelection` and observe an `isProcessing` for example, and the `BaseAddPaymentMethodFragment` would be the one that knows about the `BaseSheetViewModel`. But I started to implement too much fake behavior, so I want to have the real things before defining how this will work. I can delete this, I chose to leave so we have a structure.

Reply:
When my review is up you will want to look at the FormViewModel.  The only thing it exposes is:
```val paramMapFlow: Flow<MutableMap<String, Any?>?>``` which can be formed into the `PaymentMethodCreateParams.toParamMap()`